File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        std::set<int> num_set(nums.begin(), nums.end());

        int longest_sequence = 0;
        for (const int num : nums){
            int length = 1;
            while (num_set.count(num + length)){
                length += 1;
            }
            longest_sequence = std::max(longest_sequence, length);
        }

        return longest_sequence;

    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        counts = {}
        for num in nums:
            counts[num] = counts.get(num, 0) + 1
        
        heap = []
        for num, count in counts.items():
            heapq.heappush(heap, (-count, num))

        res = []
        while k > 0:
            count, num = heapq.heappop(heap)
            res.append(num)
            k -= 1
            
        return res


        